### DISCUSSION TITLE
Explainer give error typeError: ufunc 'isfinite' not supported for the input types

### DISCUSSION THREAD

**Discussion by farbodr** [shap/shap#2744] [Q&A]
I'm trying to create the SHAP values for a logistic regression model, and I'm getting different errors depending on what data I pass to the explainer. 

```
def calibrated_model_proba(x):
    return lr.predict_proba(x)[:,1]

explainer = shap.Explainer(calibrated_model_proba, 
                                              X_train)
shap_values = explainer(X_test)
```
This gives error

`
typeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
`

I suspect this is because X_test is being used untransformed. Passing it to calibrated_model_proba returns the correct probabilities. If I pass a transformed X_test, I get the obvious error that the transformed X_test and X_train have different shapes.

TIA,
FR

**Comment by EagleEye1107** [comment]
Any news ?

**Reply by matovitch** [reply]
I had a similar issue because `X_test` was a numpy array with dtype `object`, the type of elements in your dataset should be numeric.
